IdentityCore/src/throttling/model/MSIDThrottlingModelFactory.m

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDThrottlingModelFactory.h"
+#import "MSIDThrottlingCacheRecord.h"
+#import "MSIDLRUCache.h"
+#import "MSIDThrottlingModelBase.h"
+#import "MSIDThrottlingModelInteractionRequire.h"
+#import "MSIDThrottlingModel429.h"
+
+@implementation MSIDThrottlingModelFactory
+
++ (MSIDThrottlingModelBase *)throttlingModelForIncomingRequest:(id<MSIDThumbprintCalculatable> _Nonnull)request
+                                                   accessGroup:(NSString *)accessGroup
+                                                       context:(id<MSIDRequestContext> _Nonnull)context
+{
+    if (![MSIDThrottlingModelFactory validateInput:request]) return nil;
+    NSError *error;
```

Comment:
nit: looks like error gets unused. I'd recommend passing error has a double pointer param and then capture it as per recommend coding practice ( @oldalton fyi for your input also)

Reply:
I don't process the error because of in the case key is not existing in cache, you still throw an error. So it's weird for the calling (MSIDThrottlingModelFactory) to understand what should it do with the error.